content/administration/upgrade/odoo_sh.rst

```diff
@@ -1,66 +1,132 @@
+     upgraded database are set to be updated. If the modules are missing in the code, the update
+     fails, thus failing the whole process. An empty module with a manifest and possibly some custom
+     upgrade script are necessary to clean up the database. The complete removal of the module has
+     to be handled afterwards.
 
-* Make a new dump of your production database (as described in step 1)
-* Upload it on upgrade.odoo.com and select the Production purpose
-* Receive the newly upgraded dump and import it in your production branch. The build might get
-  marked as failed because the platform will run it with the upgraded databases' Odoo version
-  together with the old custom code.
-* Merge or commit the upgraded custom code in the production branch
+Functionally test your upgraded database
+========================================
 
-If anything goes wrong, remember you can restore a backup. The platform will always make one before
-you make any Odoo.sh operation on the production database. If the restored backup comes from a
-previous version, the platform will detect it and change the project's Odoo version back if it needs
-to.
+Now that the test upgraded database is available on your staging branch, **thoroughly test it** and
+make sure everything runs as it's supposed to. Once you are satisfied with the result, you are ready
+to upgrade your production database.
 
-Custom modules (if applicable)
-==============================
+Production upgrade
+==================
 
-The upgrade of a database that contains custom modules is a 2 step process.
+Once you are happy with your testing, you can start the process on the :guilabel:`Production`
+branch.
 
-#. The standard upgrade is done when your upgrade request is completed.
-#. Your custom modules also need to be upgraded to keep them compatible with the new version.
+On your :guilabel:`Production` branch, go to the :guilabel:`Upgrade` tab, select the
+:guilabel:`targeted version` and click on the :guilabel:`start Upgrade` button.
 
-Depending on your contract, the upgrade of your custom modules can be done
-- by yourself
-- by your Partner
-- by Odoo (if you hold a subscription to 'Maintenance of Customizations')
+.. image:: odoo_sh/odoo-sh-prod.png
+   :align: center
+   :alt: View from the upgrade tab
 
-.. seealso::
-   - :ref:`upgrade/sla`
+The actual process is **triggered as soon as you push a new commit** in your branch. Make sure you
+are pushing code that is compatible with the new version. For example by merging the code from your
+upgraded staging branch.
+
+.. note::
+   You can see the progress of the upgrade by going to the :guilabel:`Upgrade` tab of the main
+   branch.
+
+.. image:: odoo_sh/odoo-sh-progress.png
+   :align: center
+   :alt: View showing the progress of the upgrade
+
+.. important::
+   Your database is unavailable throughout the process.
+
+.. note::
+   If anything goes wrong, the platform automatically reverts the upgrade, same as for a regular
```

Comment:
I thought I had written this comment, but it seems I forgot...

"same as" alone sounds like there are a few words missing.
- similarly to...
- the same as it would be for...